Main Model/multifirm-unit-tests.nls

```diff
@@ -0,0 +1,157 @@
+__includes["unit testing.nls"]  ;including this for purpose of using the helpful
+
+
+to unit-tests
+  run-tests (list    
+    (list "Price and output tests" test-adjust-price-and-output)
+    (list "Pay wage tests" test-pay-wages)
+    (list "Fire worker tests" test-fire-workers)
+    (list "Average sales tests" test-average-sales)
+    (list "Wage rate tests" test-adjust-wage-rate)
+  )
+end
+
+;;TO IMPLEMENT: Test trends that measure the reasonableness of the model (like unemployment rate and price)
+;;We want users to be able to distinguish when the model is behaving appropriately when it might not be obvious from the output
+to test-trends
+  run-tests (list
+  )
+end
+
+
+to test-unemployment
+  setup
+  repeat 100 [
+    go
+    assert ((1 - (count employment-links / count households)) < 0.5) "Unemployment must be below 50%"
+  ]
+end
+
+to-report test-remove-firm
+  report [->
+    set BANKRUPT-FIRM-PROFITS (list)
+    create-households 1
+    create-firms 1 [
+      set firm-type 0
+    ]
+    create-firms 1 [
+      ask households [create-equity-link-with myself]
+      ask my-equity-links [set equity 100]
+      set liquidity 100
+      set firm-type 0
+      remove-firm
+    ]
+    assert (count firms = 1) "Removed firm should be replaced"
+  ]
+end
+
+
+to-report test-adjust-wage-rate
+  report [->
+    set MAX-WAGE-GROWTH 0.1
+    set MONTHS-TO-LOWER-WAGE 2
+    create-households 3
+    create-firms 1 [
+      set wage-rate 1
+      set months-without-vacancies 0
+      ask households [create-employment-link-with myself]
+      set desired-n-workers 5 
+      set fired-worker? false
+      
+      adjust-wage-rate
+      assert (wage-rate > 1) "wage should increase"
+      
+      set wage-rate 1
+      set fired-worker? true
+      adjust-wage-rate
+      assert (wage-rate < 1) "wage should decrease when we fired a worker last time period"
+      
+      set wage-rate 1
+      set months-without-vacancies 3
+      adjust-wage-rate
+      assert (wage-rate < 1) "wage should decrease when we have had a vacant position for too many months"
+      
+      set wage-rate 1
+      set months-without-vacancies 1
+      set fired-worker? false
+      adjust-wage-rate
+      assert (wage-rate = 1) "wage should not change"
+      
+    ]
+  ]
+end
+
+to-report test-adjust-price-and-output
```

Comment:
We need tests for all the possible combinations, not just price staying the same. Need to test price increasing and decreasing. 

And we need to:
- Test that ideal-production is set correctly. (that should be put into a separate procedure that gets called in adjust-output-and-price to make it easier to reason about and test)
- Test that cost of ideal-production is done correctly (this should also probably be put in a separate procedure that gets called in adjust-output-and-price)